internal/uploader.go

```diff
@@ -216,3 +222,40 @@ func (uploader *SplitStreamUploader) Clone() Uploader {
 		blockSize:  uploader.blockSize,
 	}
 }
+
+func (uploader *RegularUploader) ShowRemainingTime() {
+	defer uploader.waitGroup.Done()
+	startTime := time.Now()
+	e := ewma.SimpleEWMA{}
+	e.Add(0)
+	for {
+		totalSizeInt, err := uploader.RawDataSize()
+		if err != nil {
+			return
+		}
+		uploadedSizeInt, err := uploader.UploadedDataSize()
+		if err != nil {
+			return
+		}
+
+		totalSize := float64(totalSizeInt) / 1000000
+		uploadedSize := float64(uploadedSizeInt) / 1000000
+
+		remainingSize := totalSize - uploadedSize
```

Comment:
It looks like you are trying to compare different sizes:
AFAIK `RawDataSize()` is uncompressed size
`UploadedDataSize()` is compressed & encrypted size that was uploaded

So, when we have compression enabled it should never meet each other and Remaining time should only grow.

Isn't it?

Reply:
tried to solve this problem

Reply:
I don't think that it will work: `compressedFile` is `io.Reader`... You are trying to read-and-copy data from `io.Reader` to `bytes.Buffer{}`... and then upload.
However you can read from `io.Reader` only once... so, I think that this code will corrupt data you are trying to upload.

Reply:
I'm not really comfortable about current sollution, but it is the best my brain can afford now...